src/main/java/com/example/springboot/service/BookServiceImpl.java

```diff
@@ -62,4 +64,13 @@ public List<BookDto> search(BookSearchParameters params) {
                 .map(bookMapper::toDto)
                 .toList();
     }
+
+    @Override
+    public List<BookDtoWithoutCategoryIds> findAllByCategory(Long id, Pageable pageable) {
+        Category category = new Category();
+        category.setId(id);
```

Comment:
Have you checked that Id is valid? (category with such id exist)
The preferred way to `categoryRepo.findCategoryById(id)` and process as optional.